tests/datasource/data_connector/test_sql_data_connector.py

```diff
@@ -1103,6 +1104,111 @@ def test_more_complex_instantiation_of_ConfiguredAssetSqlDataConnector(
     }
 
 
+def test_more_complex_instantiation_of_ConfiguredAssetSqlDataConnector_include_schema_name(
+    test_cases_for_sql_data_connector_sqlite_execution_engine,
+):
+    my_data_connector = instantiate_class_from_config(
+        config={
+            "class_name": "ConfiguredAssetSqlDataConnector",
+            "name": "my_sql_data_connector",
+            "assets": {
+                "table_partitioned_by_date_column__A": {
+                    "splitter_method": "_split_on_column_value",
+                    "splitter_kwargs": {"column_name": "date"},
+                    "include_schema_name": True,
+                    "schema_name": "main",
+                },
+            },
+        },
+        runtime_environment={
+            "execution_engine": test_cases_for_sql_data_connector_sqlite_execution_engine,
+            "datasource_name": "my_test_datasource",
+        },
+        config_defaults={"module_name": "great_expectations.datasource.data_connector"},
+    )
+    assert "main.table_partitioned_by_date_column__A" in my_data_connector.assets
+
+    # schema_name given but include_schema_name is set to False
+    with pytest.raises(ge_exceptions.DataConnectorError):
+        my_data_connector = instantiate_class_from_config(
+            config={
+                "class_name": "ConfiguredAssetSqlDataConnector",
+                "name": "my_sql_data_connector",
+                "assets": {
+                    "table_partitioned_by_date_column__A": {
+                        "splitter_method": "_split_on_column_value",
+                        "splitter_kwargs": {"column_name": "date"},
+                        "schema_name": "main",
+                        "include_schema_name": False,
+                    },
+                },
+            },
+            runtime_environment={
+                "execution_engine": test_cases_for_sql_data_connector_sqlite_execution_engine,
+                "datasource_name": "my_test_datasource",
+            },
+            config_defaults={
+                "module_name": "great_expectations.datasource.data_connector"
+            },
+        )
+
+
+def test_more_complex_instantiation_of_ConfiguredAssetSqlDataConnector_include_schema_name_prefix_suffix(
+    test_cases_for_sql_data_connector_sqlite_execution_engine,
+):
+    my_data_connector = instantiate_class_from_config(
```

Comment:
If we know the type in advance, can an argument be made to just use the class constructor? I don't think this is strictly necessary but I know that this particular method is a bit costly/complex.

Reply:
Same note throughout these tests

Reply:
Great point. I'll add these changes now.